relay-general/src/store/transactions/rules.rs

```diff
@@ -94,6 +94,70 @@ impl Default for RedactionRule {
     }
 }
 
+/// The rule describes how span descriptions should be changed.
+#[derive(Debug, Clone)]
+pub struct SpanDescriptionRule {
+    /// The pattern which will be applied to the span description.
+    pub pattern: LazyGlob,
+    /// Date time when the rule expires and it should not be applied anymore.
+    pub expiry: DateTime<Utc>,
+    /// Object containing transaction attributes the rules must only be applied to.
+    pub scope: RuleScope,
+    /// Object describing what to do with the matched pattern.
+    pub redaction: RedactionRule,
+}
+
+impl From<&TransactionNameRule> for SpanDescriptionRule {
+    fn from(value: &TransactionNameRule) -> Self {
+        SpanDescriptionRule {
+            pattern: LazyGlob::new(format!("**{}", value.pattern.raw)),
+            expiry: value.expiry,
+            scope: value.scope.clone(),
+            redaction: value.redaction.clone(),
+        }
+    }
+}
+
+impl SpanDescriptionRule {
+    pub fn match_and_apply(&self, mut string: Cow<String>) -> Option<String> {
```

Comment:
I think we could benefit from having a trait for these rules. I'll look into this at some point.